test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,266 @@
+	return m.oc.ExecWithoutExit(m.PodName, "mysql -h "+"127.0.0.1"+" -u"+m.Env["MYSQL_USER"]+" -p"+m.Env["MYSQL_PASSWORD"]+" -e \""+sql+"\" "+m.Env["MYSQL_DATABASE"]).Verbose().Output()
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) WaitForMySQLOutput(timeout time.Duration, root bool, sql, resultSubstr string) error {
+	return wait.Poll(5*time.Second, timeout, func() (bool, error) {
+		out, err := m.QueryAsRoot(sql)
+		if err != nil {
+			return false, err
+		}
+		if strings.Contains(out, resultSubstr) {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+var _ = g.Describe("images: MySQL replication template", func() {
+	defer g.GinkgoRecover()
+	var (
+		volumeTmpDir string
+		templatePath = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		oc           = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user         = "user"
+		userPass     = "userpass"
+		rootPass     = "rootpass"
+		database     = "userdb"
+	)
+
+	g.Describe("MySQL replication", func() {
+		g.JustBeforeEach(func() {
+			var err error
+			volumeTmpDir, err = ioutil.TempDir(exutil.TestContext.OutputDir, "persistent-volume")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = os.Chmod(volumeTmpDir, 0777)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			pv := exutil.CreatePersistentVolume("mysql-pv", "1Gi", volumeTmpDir)
+			_, err = oc.AdminKubeREST().PersistentVolumes().Create(pv)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.AfterEach(func() {
+			err := oc.AsAdmin().Run("delete").Args("pods", "--all").Verbose().Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().Run("delete").Args("all", "--all").Verbose().Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().Run("delete").Args("pvc", "--all").Verbose().Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.AsAdmin().Run("delete").Args("pv", "mysql-pv").Verbose().Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.It("should work", func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("processing and creating the template %q", templatePath))
+			configFile, err := oc.Run("process").Verbose().
+				Args(
+				"-f", templatePath,
+				"-v", "MYSQL_USER="+user+",MYSQL_PASSWORD="+userPass+",MYSQL_ROOT_PASSWORD="+rootPass+",MYSQL_DATABASE="+database).
+				OutputToFile("config.json")
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = oc.Run("create").Args("-f", configFile).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for mysql-master endpoint")
+			err = oc.KubeFramework().WaitForAnEndpoint("mysql-master")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for mysql-slave endpoint")
+			err = oc.KubeFramework().WaitForAnEndpoint("mysql-slave")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// Get pod names
+			podNames, err := getPods(oc, selector("name=mysql-master"))
```

Comment:
if this returns just names for pods, the function should be called "getPodsNames"